src/components/menu/provider.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, Dispatch, FC, HTMLAttributes, SetStateAction, useState } from 'react'
+
+interface MenuContextValue {
+  triggerProps: HTMLAttributes<HTMLButtonElement>
+  popoverProps: HTMLAttributes<HTMLDivElement>
+  isOpen: boolean
+  setIsOpen: Dispatch<SetStateAction<boolean>>
```

Comment:
**suggestion:** Personally, I think functions like `open` and `close` are preferable to `setIsOpen` because there's a higher cognitive load for the reader when they see `setIsOpen(true)` or `setIsOpen(false)` as compared to reading `open` or `close`